content/chainguard/chainguard-enforce/chainguard-enforce-kubernetes/detect-log4shell-demo/index.md

```diff
@@ -86,45 +85,53 @@ As you did with the IAM group, copy the `NAME` value of the cluster you want to
-cosign download sbom ghcr.io/chainguard-dev/log4shell-demo/app:v0.1.0
-```
+You're now ready to test how Chainguard Enforce works by deploying a sample application that includes a vulnerable version of Log4j. You'll then create a policy that will detect this vulnerability.
 
-Be aware that this command returns a warning letting you know that it is only meant to show the contents of an image, not its provenance. Since we know this image comes directly from Chainguard, there's no need for this warning to concern us.
+The sample container image used in this step is maintained by Chainguard for demonstration purposes only. Because this image includes a vulnerable version of Log4j, you _**should not run this image in a production environment.**_ The procedure outlined in this tutorial will have you create a policy that will cause Chainguard Enforce to block a container built from this image from running.
 
-When you're ready, run a pod in the tenant cluster using this sample container image.
+If you'd like, you can inspect the image's [SBOM](https://edu.chainguard.dev/software-security/glossary/#sbom) by retrieving it directly from the container registry with the following `cosign` command. If you don’t already have `cosign` installed, you can install it following our [How to Install Cosign guide](https://edu.chainguard.dev/open-source/sigstore/cosign/how-to-install-cosign/).
 
 ```sh
-kubectl run log4shell-demo --image=ghcr.io/chainguard-dev/log4shell-demo/app:v0.1.0
+cosign download attestation ghcr.io/chainguard-dev/log4shell-demo/app:v0.1.0 | jq -r .payload | base64 -d | jq > log4shell-sbom.json
 ```
 
-Chainguard Enforce is only able to validate resources in namespaces that have chosen to opt-in. This can be done by adding the label `policy.sigstore.dev/include: "true"` to the namespace resource.
+This command downloads an attestation containing the image's SBOM from the same location where the image is found. However, this SBOM data is encoded in base64, making it unreadable without further processing. This is why the output from the first part of the command is piped into `jq` in order to filter out the `payload` section of the output containing the SBOM. 
+
+This filtered output is then passed into the `base64` command to be decoded before that output is piped into another `jq` command. This final `jq` command processes the decoded JSON and writes it to a file named `log4shell-sbom.json`.
+
+Following that, you can inspect the SBOM by reading the content of this new file.
 
 ```sh
-kubectl label ns default policy.sigstore.dev/include=true --overwrite
+cat log4shell-sbom.json
 ```
 
-Following this, the image will appear in the existence records. To verify, list the cluster's records, grepping for “log4shell”.
+When you're ready, run a pod in the tenant cluster using this sample container image.
 
 ```sh
-chainctl cluster records list $CLUSTER | grep log4shell
+kubectl run log4shell-demo --image=ghcr.io/chainguard-dev/log4shell-demo/app:v0.1.0
```

Comment:
I thought it may be helpful to explain that you are just now setting up the `log4shell-demo` pod here and that you'll remove it later.

Reply:
Oooh, you don't actually run the pod since Enforce blocks it. Good catch.